### ISSUE TITLE
[Question] Query parameter for Azure B2C

### ISSUE THREAD

**Issue by Ponant** [AzureAD/microsoft-identity-web#1458]
Hello,
I need to serve pages in French and English to users, and have B2C flows translated and available in both languages. 
Currently, users are provided with different languages on the b2clogin.com signin/up pages provided I add
`&lang=fr` as a query string to the b2clogin.com endpoint.
Can anyone guide a bit on how/where to add the query when using `AddMicrosoftIdentityWebApp` in a Razor Page web app (with cookie auth)?
Thank you!

**Comment by jmprieur** [comment]
@Ponant

I would think that you could add from the OnRedirectToIdentityProvider event:

```CSharp
services.Configure<OpenIdConnectOptions>(OpenIdConnectDefaults.AuthenticationScheme, options =>
{
    var redirectToIdpHandler = options.Events.OnRedirectToIdentityProvider;
    options.Events.OnRedirectToIdentityProvider = async context =>
    {
     await redirectToIdpHandler(context);
     context.ProtocolMessage.SetParameter("lang", "fr");
    }

});
```

**Comment by jmprieur** [comment]
You're right @Ponant. you can also use the override of AddMicrosoftIdentityWebApp  that takes delegates. 
But I'm too lazy to write the configuration.Bind() :-), so I prefer the service.Configure approach.
